
\begin{table}[hb!]
\centering
  
\begin{tabular}{ p{\textwidth } }
  \hline
  
  \textbf{Decay of Differential Expression} \\
  \hline

 
  \begin{description}

    \item[\textbf{Input}]: Define the gene expression data and the network model

    \item [Calculate] the moderated $t$-statistic $g_k$ between cases and
          controls of each gene: $g_1, g_2,\dots,g_{N}$
          % (where $N$ is the number of genes both assayed and on the network)

    \item[For] putative driver gene $i$ 

    \begin{enumerate}

    	\item Compute the observed discordance between differential expression and distance from gene $i$: $\tau_{B}( \{g_1,\dots,g_M\}, \{d(1,i),\dots,d(M,i)\}$
      where the gene-level statistics are indexed so that the first $M$ genes are in the neighborhood of gene $i$, $\tau_B$ is the Kendall rank correlation,
      and $d(k,i)$ is the geodesic distance between gene $k$ and gene $i$


      \item Compute the distribution of the discordance under the null by
      generating $N_{\text{Decay}}$ null discordances. To
      generate each null discordance:

      \begin{enumerate}

          \item Permute the phenotype labels and recalculate the moderated
          $t$-statistics for each gene

          \item Recompute the discordance as in Step (1)

      \end{enumerate}


      \item Compute $p^i_{\text{Decay}}$ by counting the the proportion of 
            null discordances less than or equal to the observed discordance. 
            If $p^i_{\text{Decay}} = 0$, set 
            $p^i_{\text{Decay}}=\frac{1}{N_\text{Decay}+1}$

    \end{enumerate}


    \item[\textbf{Output}]: Return $p^i_{\text{Decay}}$


  \end{description} \\
  
  \hline
\end{tabular}

\caption{The Decay of Differential Expression procedure tests if the discordance
between differential expression and distance from a putative driver gene is
greater with the phenotype labels we observe than with a random permutation 
of the labels.}

\label{tab:procedure-decay}
\end{table}